server/app/com/xsn/explorer/data/anorm/dao/BlockRewardPostgresDAO.scala

```diff
@@ -6,80 +6,108 @@ import anorm._
 }
 import com.xsn.explorer.models.values.Blockhash
-import com.xsn.explorer.data.anorm.parsers.BlockRewardParsers.parseBlockReward
+import com.xsn.explorer.data.anorm.parsers.BlockRewardParsers.{parseBlockReward, parseSummary}
+import com.xsn.explorer.models.rpc.TransactionVIN
 
 class BlockRewardPostgresDAO {
 
   def upsert(blockhash: Blockhash, reward: BlockRewards)(implicit conn: Connection): Unit = {
-    getRewards(reward).foreach(r => upsert(blockhash, r._1, r._2))
+    getRewards(reward).foreach(r => upsert(blockhash, r._1, r._2, r._3))
   }
 
   def getBy(blockhash: Blockhash)(implicit conn: Connection): Option[BlockRewards] = {
     val rewards = getRewards(blockhash)
     rewards match {
-      case (r, RewardType.PoW) :: Nil =>
+      case (r, RewardType.PoW, _) :: Nil =>
         Some(PoWBlockRewards(r))
-      case (r, RewardType.PoS) :: Nil =>
-        Some(PoSBlockRewards(r, None))
-      case (r, RewardType.PoS) :: (masternode, RewardType.Masternode) :: Nil =>
-        Some(PoSBlockRewards(r, Some(masternode)))
-      case (owner, RewardType.TPoSOwner) :: (merchant, RewardType.TPoSMerchant) :: Nil =>
-        Some(TPoSBlockRewards(owner, merchant, None))
-      case (masternode, RewardType.Masternode) :: (owner, RewardType.TPoSOwner) :: (merchant, RewardType.TPoSMerchant) :: Nil =>
-        Some(TPoSBlockRewards(owner, merchant, Some(masternode)))
+      case (r, RewardType.PoS, Some(stakedVIN)) :: Nil =>
+        Some(PoSBlockRewards(r, None, stakedVIN))
+      case (r, RewardType.PoS, Some(stakedVIN)) :: (masternode, RewardType.Masternode, _) :: Nil =>
+        Some(PoSBlockRewards(r, Some(masternode), stakedVIN))
+      case (owner, RewardType.TPoSOwner, Some(stakedVIN)) :: (merchant, RewardType.TPoSMerchant, _) :: Nil =>
+        Some(TPoSBlockRewards(owner, merchant, None, stakedVIN))
+      case (masternode, RewardType.Masternode, _) :: (owner, RewardType.TPoSOwner, Some(stakedVIN))
+            :: (merchant, RewardType.TPoSMerchant, _) :: Nil =>
+        Some(TPoSBlockRewards(owner, merchant, Some(masternode), stakedVIN))
       case Nil => None
       case _ =>
         throw new RuntimeException("Unknown reward type")
     }
   }
 
-  def deleteBy(blockhash: Blockhash)(implicit conn: Connection): List[(BlockReward, RewardType)] = {
+  def deleteBy(
+      blockhash: Blockhash
+  )(implicit conn: Connection): List[(BlockReward, RewardType, Option[TransactionVIN.Raw])] = {
     SQL(
       """
         |DELETE FROM block_rewards
         |WHERE blockhash = {blockhash}
-        |RETURNING address, value, type
+        |RETURNING address, value, type, staked_txid, staked_index
       """.stripMargin
     ).on(
         'blockhash -> blockhash.toBytesBE.toArray
       )
       .as(parseBlockReward.*)
   }
 
-  private def upsert(blockhash: Blockhash, reward: BlockReward, rewardType: RewardType)(
+  def getSummary(numberOfBlocks: Int)(implicit conn: Connection): BlockRewardsSummary = {
+    SQL(
+      """
+        |select
```

Comment:
Please be consistent with the existing syntax, reserved SQL words are upper case, column names snake_case